aodntools/timeseries_products/aggregated_timeseries.py

```diff
@@ -331,26 +327,29 @@ def main_aggregator(files_to_agg, var_to_agg, site_code, input_dir='', output_di
     NOMINAL_DEPTH = ds.createVariable(varname='NOMINAL_DEPTH', **inst_float_template)
 
     ## main loop
+    start = 0
     for index, file in enumerate(files_to_agg):
         with xr.open_dataset(os.path.join(input_dir, file)) as nc:
             nc = in_water(nc)
-            start = sum(varlen_list[:index + 1])
-            end = sum(varlen_list[:index + 2])
+            n_obs = len(nc.TIME)
+            end = start + n_obs
```

Comment:
rename n_obs to total_nobs above, yo are mutating n_obs here. I would also use slices:
```
nc = in_water(nc)
chunk = slice(start,start+n_obs);
DEPTH[chunk] ...
...
start = n_obs;
```

In fact, this ```with``` block shoudl be a function receiving ```ds,dsvars,nc```, after you define some top-level constants (one_day,epoch,etc).

Reply:
Also, you could store the ```len(nc.TIME)``` of the files from the loop above (278)) and use it here in the for iterator.

Reply:
I've renamed to `n_obs_total` above. The rest I don't really see the point, sorry. Anyway this is not meant to be a refactoring PR, so I'll try to keep these changes minimal.